src/extension.ts

```diff
@@ -280,7 +283,7 @@ function initAsyncCommand<T>(extensionContext: vscode.ExtensionContext, commandI
             if (args.length === 0) {
                 await callback();
             } else {
-                await callback(<T>args[0]);
+                await callback.apply(null, args);
```

Comment:
I'm not sure setting `this` to `null` will work in all cases? What if the callback has a different value they want for `this`?

Reply:
I guess that the caller has to remember to use arrow functions all the time. Otherwise, is that anyway to maintain that?

Reply:
`callback.apply(null, args)` is equivalent to `callback(...args)`.  If caller wants a specific this context when the callback is invoked, they need to do the binding in the callback.  e.g.,

```
callback = function() {
    myFunc.apply(myThis, args);
}
```

Reply:
Yeah, was wondering how TS transcompile would avoid this problem.

I guess after this change, it requires all callers to remember always use arrow functions as callback to preserve `this`. Does it make sense?

Reply:
Yeah, either arrow functions, or explicitly bound functions via [Function's bind() method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind), or equivalent as above.